boa/src/profiler.rs

```diff
@@ -0,0 +1,103 @@
+#![allow(missing_copy_implementations, missing_debug_implementations)]
+
+#[cfg(feature = "profiler")]
+use measureme::{EventId, Profiler, TimingGuard};
+#[cfg(feature = "profiler")]
+use once_cell::sync::OnceCell;
+use std::fmt::{self, Debug};
+#[cfg(feature = "profiler")]
+use std::{
+    path::Path,
+    thread::{current, ThreadId},
+};
+
+/// MmapSerializatioSink is faster on macOS and Linux
+/// but FileSerializationSink is faster on Windows
+#[cfg(not(windows))]
+#[cfg(feature = "profiler")]
+type SerializationSink = measureme::MmapSerializationSink;
+#[cfg(windows)]
+#[cfg(feature = "profiler")]
+type SerializationSink = measureme::FileSerializationSink;
+#[cfg(feature = "profiler")]
+pub struct BoaProfiler {
+    profiler: Profiler<SerializationSink>,
+}
+
+#[cfg(feature = "profiler")]
+pub static mut INSTANCE: OnceCell<BoaProfiler> = OnceCell::new();
```

Comment:
Does this need to be `pub`, considering that it is accessed safely via `BoaProfiler::global()`?

Reply:
Thanks @brian-gavin thats correct, changed now

Reply:
Thanks!